#!/usr/bin/env python
PACKAGE = "sentry_chassis_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name       Type      Level Description        Default  Min   Max
# --- Drive Motor (轮子速度) PID ---
gen.add("drive_p", double_t, 0,    "Drive P Gain",    5.0,     0.0,  100.0)
gen.add("drive_i", double_t, 0,    "Drive I Gain",    0.01,    0.0,  10.0)
gen.add("drive_d", double_t, 0,    "Drive D Gain",    0.0,     0.0,  10.0)

# --- Pivot Motor (舵向角度) PID ---
gen.add("pivot_p", double_t, 0,    "Pivot P Gain",    10.0,    0.0,  100.0)
gen.add("pivot_i", double_t, 0,    "Pivot I Gain",    0.0,     0.0,  10.0)
gen.add("pivot_d", double_t, 0,    "Pivot D Gain",    0.5,     0.0,  10.0)

# 退出并生成名为 SentryPidsConfig 的 C++ 头文件
exit(gen.generate(PACKAGE, "sentry_chassis_controller", "SentryPids"))
